Add new Dockerfiles targeting ruby 3.4

MatthewPickinYork opened the pull request:
- Make 3.4-specific changes to Dockerfiles, where applicable
- Fetch 3.4 lambda layer from mjp513 sandbox AWS account

fd-2030

MatthewPickinYork committed:
Add new Dockerfiles targeting ruby 3.4

- Make 3.4-specific changes to Dockerfiles, where
  applicable
- Fetch 3.4 lambda layer from mjp513 sandbox AWS
  account

fd-2030

MatthewPickinYork committed:
Add 3.4 images to build workflow matrixes

fd-2030

njm506 reviewed (approved):
Yup, I can see that the package lists are similar and the changes from 3.3 are minimal

MatthewPickinYork committed:
Specify test lambda role and layer

- Update image build workflow to use custom
  sandbox IAM role (instead of SYS-provided one)
- For 3.4 ruby-postgres image dependency script, fetch
  layer from sandbox account

fd-2030

MatthewPickinYork committed:
Bump layer version

fd-2030

MatthewPickinYork committed:
Split Ruby 3.4 image build process into own job

fd-2030

MatthewPickinYork committed:
Remove Ruby 3.3 build completely

fd-2030
